scripts/ci/reborn_changed_coverage.py

```
@@ -38,6 +57,73 @@ class GateError(RuntimeError):
+        """
+
+        if not path.startswith(f"{CRATES_ROOT}/") or not path.endswith(".rs"):
+            return
+        if self.owner(path) is not None:
+            return
+        raise GateError(
+            f"changed Rust file under {CRATES_ROOT}/ belongs to no discovered crate: "
+            f"{path}. Refusing to treat it as 'not production' — an unattributable "
+            "path means the crate inventory and the tree disagree "
+            "(docs/reborn/target-architecture/CHECKLIST.md WS10)."
+        )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)